src/config.rs

```diff
@@ -1,64 +1,184 @@
-use super::internal::{Bounded, Infinite, SizeLimit};
-use byteorder::{BigEndian, ByteOrder, LittleEndian, NativeEndian};
+use byteorder;
 use de::read::BincodeRead;
-use error::Result;
+use error::{ErrorKind, Result};
 use serde;
 use std::io::{Read, Write};
 use std::marker::PhantomData;
-use {DeserializerAcceptor, SerializerAcceptor};
+
+pub(crate) use self::internal::*;
 
 use self::EndianOption::*;
 use self::LimitOption::*;
 
-struct DefaultOptions(Infinite);
-
-pub(crate) trait Options {
-    type Limit: SizeLimit + 'static;
-    type Endian: ByteOrder + 'static;
-
-    fn limit(&mut self) -> &mut Self::Limit;
-}
+/// The default options for bincode serialization/deserialization.
+/// Implements OptionsExt to allow building configuration object for non-default settings.
+///
+/// ### Defaults
+/// By default bincode will use little-endian encoding for mult-byte integers, and will not
+/// limit the number of serialized/deserialized bytes.
+#[derive(Copy, Clone)]
+pub struct DefaultOptions(Infinite);
```

Comment:
Nitpick: DefaultOptions is defined here, below this is the `OptionsExt` trait, and below that is `impl DefaultOptions`